### PR TITLE
[release/7.0][workloads] Backport UsingMobileWorkload change

### PR DESCRIPTION
**Pull request by steveisok**
Backport of https://github.com/dotnet/runtime/pull/77489

Adds the UsingMobileWorkload property along with a tighter check on when to import the aot task

### CODE REVIEW

**Review by lewing — COMMENTED** [review]
~doesn't this code already live in Current?~ I missed this was targeting `release/7.0`

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.net7.Manifest/WorkloadManifest.targets.in [hunk 1]
```diff
@@ -38,7 +38,11 @@
       <UsingBlazorAOTWorkloadManifest>true</UsingBlazorAOTWorkloadManifest>
     </PropertyGroup>
 
-    <Import Condition="'$(TargetsNet7)' == 'true' and '$(RunAOTCompilation)' == 'true' and '$(_BrowserWorkloadDisabled7)' != 'true'" Project="Sdk.props" Sdk="Microsoft.NET.Runtime.MonoAOTCompiler.Task.net7" />
+    <PropertyGroup Condition="'$(TargetPlatformIdentifier)' == 'android' or '$(TargetPlatformIdentifier)' == 'macOS' or '$(TargetPlatformIdentifier)' == 'maccatalyst' or '$(TargetPlatformIdentifier)' == 'ios' or '$(TargetPlatformIdentifier)' == 'tvos'">
+      <UsingMobileWorkload>true</UsingMobileWorkload>
+    </PropertyGroup>
+
```
Comment (lewing) [inline src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.net7.Manifest/WorkloadManifest.targets.in:44]:
```suggestion
```

**Comment by lewing** [comment]
should add the fixed condition to net6 too

**Review by lewing — APPROVED** [review]

**Comment by lewing** [comment]
wasm failure is https://github.com/dotnet/runtime/issues/82271

**Comment by steveisok** [comment]
@carlossanlop this is a tell mode change.

**Comment by mmitche** [comment]
> @mmitche are you ok with merging this? Just want to confirm that this won't mess with the build flows with the servicing branch closed.

Please wait until branding/checkins tomorrow.

**Pull request merged**